src/eko/io/runcards.py

```diff
@@ -309,7 +309,7 @@ def flavored_mugrid(mugrid: list, masses: list, matching_ratios: list):
 
     """
     atlas = default_atlas(masses, matching_ratios)
-    return [(mu, nf_default(mu**2, atlas)) for mu in mugrid]
+    return [(float(mu), nf_default(mu**2, atlas)) for mu in mugrid]
```

Comment:
Why do you explicitly need to call `float`?

It is not explicit in the type above, but I thought `mugrid` as a `list[float]`.

Reply:
It's because otherwise we had problems with the dumping into a .yaml since it was a numpy object

Reply:
After https://github.com/NNPDF/eko/blob/633fbb48f6a5d85b49757f7417c1c0ca4b5f16ff/src/eko/io/runcards.py#L223 elements are numpy-objects (so tuple for yaml)

Reply:
But that's an actual array, so it is not a `list` at all.

The function 
https://github.com/NNPDF/eko/blob/633fbb48f6a5d85b49757f7417c1c0ca4b5f16ff/src/eko/io/runcards.py#L224-L228
should be simply called with `mugrid.tolist()`, instead of just `mugrid` (my bad, most likely I changed the function afterward - we definitely need `mypy` running...).

Reply:
yes - after that (my) line `mugrid` is an numpy-array containing numpy objects which then via your snippet arrive to https://github.com/NNPDF/eko/blob/633fbb48f6a5d85b49757f7417c1c0ca4b5f16ff/src/eko/io/runcards.py#L312
and so the observable card receives a list of tuples with the first element a numpy float and the second an integer, because there we already have the explicit cast in place: https://github.com/NNPDF/eko/blob/633fbb48f6a5d85b49757f7417c1c0ca4b5f16ff/src/eko/matchings.py#L188

Reply:
(it seems this was detected by unit testing, but not this new change ...)

Reply:
@AleCandido are you happy with this explanation?

Reply:
Sorry, I haven't replied before: I have nothing against the explanation, only something against the present situation.

Can we just call `.tolist()` there and normalize it?

Reply:
Have a look to https://github.com/NNPDF/eko/pull/274/commits/a500ddccd22d5a85acf9db9686a2193c0f96954e - the new added test fails if we're not doing anything